# Tutorial: Palmer Penguins Dataset

[short description]

In [12]:
# Load Penguins dataset
import pandas as pd
from tidyversetopandas import tidyversetopandas as ttp
penguins = pd.read_csv("penguins.csv")
penguins.head()

rowid species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      1  Adelie  Torgersen            39.1           18.7              181.0   
1      2  Adelie  Torgersen            39.5           17.4              186.0   
2      3  Adelie  Torgersen            40.3           18.0              195.0   
3      4  Adelie  Torgersen             NaN            NaN                NaN   
4      5  Adelie  Torgersen            36.7           19.3              193.0   

   body_mass_g     sex  year  
0       3750.0    male  2007  
1       3800.0  female  2007  
2       3250.0  female  2007  
3          NaN     NaN  2007  
4       3450.0  female  2007

## Remove NAs

Let’s start by finding out how many na value in our columns. 

In [6]:
penguins.isna().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

There are 11 case of `na` in sex, lets try to remove them with `filter` function in `ttp` and also build in function of `isnull` from pandas.

In [15]:
newPenguins = ttp.filter(penguins, "~ sex.isnull()")
newPenguins.isna().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

Looks great! We successfully removed all `na` from sex.

## Filter species and size

Next, we want lets limit our study to penguins with `species` of "Adelie" and `body_mass_g` bigger than 3000 gram. And function `filter` is perfect fot this job. We started with 333 rows and we should see a decrease in number of rows after filter

In [16]:
newPenguins.shape

(333, 9)

In [17]:
newPenguins = ttp.filter(newPenguins, "species == 'Adelie' & body_mass_g > 3000")
newPenguins.shape

(138, 9)

We do see a reduce of rows to 138 which is a great sign. Lets check the dataframe to make sure only "Adelie" penguins avaliable and size larger than 3000 grams. 

In [18]:
newPenguins.species.unique()

array(['Adelie'], dtype=object)

In [19]:
newPenguins.body_mass_g.min()

3050.0

We did it again! There is only "Adelie" penguins in species and the smallest penguins has size of 3050 grams.

## Add new column: BMI or smthg (mutate)

## Select only some columns (select)

## Arrange by some column (arrange)

## Do everything with pipe (all the above)